### PR TITLE
*: add docs.

### PR DESCRIPTION
**Pull request by siddontang**
PTAL @overvenus @queenypingcap

### CODE REVIEW

File: src/counter.rs [hunk 1]
```diff
@@ -29,11 +29,13 @@ pub struct Counter {
 }
 
 impl Counter {
+    /// `new` creates a counter with `name` and `help`.
```
Comment (queenypingcap) [inline src/counter.rs]: /// `new` creates a counter with the `name` and `help` arguments.

File: src/counter.rs [hunk 2]
```diff
@@ -29,11 +29,13 @@ pub struct Counter {
 }
 
 impl Counter {
+    /// `new` creates a counter with `name` and `help`. 
     pub fn new<S: Into<String>>(name: S, help: S) -> Result<Counter> {
         let opts = Opts::new(name, help);
         Counter::with_opts(opts)
     }
 
+    /// `with_opts` creates a counter with option `opts`.
```
Comment (queenypingcap) [inline src/counter.rs]: /// `with_opts` creates a counter with the `opts` option.

File: src/histogram.rs [hunk 3]
```diff
@@ -78,37 +78,44 @@ pub struct HistogramOpts {
 }
 
 impl HistogramOpts {
+    /// `new` creates a HistogramOpts with `name` and `help`.
```
Comment (queenypingcap) [inline src/histogram.rs]: /// `new` creates a HistogramOpts with the `name` and `help` arguments.

File: src/histogram.rs [hunk 4]
```diff
@@ -78,37 +78,44 @@ pub struct HistogramOpts {
...
     pub fn new<S: Into<String>>(name: S, help: S) -> HistogramOpts {
         HistogramOpts {
             common_opts: Opts::new(name, help),
             buckets: Vec::from(DEFAULT_BUCKETS as &'static [f64]),
         }
     }
 
+    /// `namespace` sets namespace.
```
Comment (queenypingcap) [inline src/histogram.rs]: /// `namespace` sets the namespace.

File: src/histogram.rs [hunk 5]
```diff
@@ -78,37 +78,44 @@ pub struct HistogramOpts {
...
     pub fn namespace<S: Into<String>>(mut self, namesapce: S) -> Self {
         self.common_opts.namespace = namesapce.into();
         self
     }
 
+    /// `sub_system` sets sub system.
```
Comment (queenypingcap) [inline src/histogram.rs]: /// `sub_system` sets the sub system.

File: src/histogram.rs [hunk 6]
```diff
@@ -78,37 +78,44 @@ pub struct HistogramOpts {
...
         self.common_opts = self.common_opts.const_label(name, value);
         self
     }
 
+    /// `fq_name` returns the fq_name.
     pub fn fq_name(&self) -> String {
         self.common_opts.fq_name()
     }
 
+    /// `buckets` set the buckets.
```
Comment (queenypingcap) [inline src/histogram.rs]: `buckets` sets the buckets.

File: src/histogram.rs [hunk 7]
```diff
@@ -195,6 +202,7 @@ pub struct Histogram {
 }
 
 impl Histogram {
+    /// `with_opts` creates a Histogram with options `opts`.
```
Comment (queenypingcap) [inline src/histogram.rs]: /// `with_opts` creates a Histogram with the `opts` options.

File: src/histogram.rs (hunk 6 above)
Comment (queenypingcap) [inline src/histogram.rs]: /// `buckets` sets the buckets.

File: src/histogram.rs [hunk 8]
```diff
@@ -225,6 +233,7 @@ impl Histogram {
 }
 
 impl Histogram {
+    /// `observe` adds a single observation to the histogram.
```
Comment (queenypingcap) [inline src/histogram.rs]: histogram -> Histogram.

File: src/histogram.rs [hunk 9]
```diff
@@ -279,6 +288,7 @@ impl MetricVecBuilder for HistogramVecBuilder {
 pub type HistogramVec = MetricVec<HistogramVecBuilder>;
 
 impl HistogramVec {
+    /// `new` creates a HistogramVec with options `opts` and label names `label_names`.
```
Comment (queenypingcap) [inline src/histogram.rs]: /// `new` creates a HistogramVec with the `opts` options and the `label_names` label names.

File: src/metrics.rs [hunk 10]
```diff
@@ -72,6 +76,7 @@ pub struct Opts {
 }
 
 impl Opts {
+    /// `new` creates the Opts with `name` and `help`.
```
Comment (queenypingcap) [inline src/metrics.rs]: /// `new` creates the Opts with the `name` and `help` arguments.

File: src/metrics.rs [hunk 11]
```diff
@@ -82,26 +87,31 @@ impl Opts {
         }
     }
 
+    /// `namespace` sets namespace.
```
Comment (queenypingcap) [inline src/metrics.rs]: /// `namespace` sets the namespace.

File: src/metrics.rs [hunk 12]
```diff
@@ -82,26 +87,31 @@ impl Opts {
...
     pub fn namespace<S: Into<String>>(mut self, namesapce: S) -> Self {
         self.namespace = namesapce.into();
         self
     }
 
+    /// `sub_system` sets sub system.
```
Comment (queenypingcap) [inline src/metrics.rs]: /// `sub_system` sets the sub system.

File: src/metrics.rs [hunk 13]
```diff
@@ -82,26 +87,31 @@ impl Opts {
...
     pub fn sub_system<S: Into<String>>(mut self, sub_system: S) -> Self {
         self.sub_system = sub_system.into();
         self
     }
 
+    /// `const_labels` sets const labels.
```
Comment (queenypingcap) [inline src/metrics.rs]: /// `const_labels` sets the const labels.

**Comment by siddontang** [comment]
PTAL @queenypingcap @overvenus

File: src/histogram.rs [hunk 14]
```diff
@@ -279,6 +288,8 @@ impl MetricVecBuilder for HistogramVecBuilder {
 pub type HistogramVec = MetricVec<HistogramVecBuilder>;
 
 impl HistogramVec {
+    /// `new` creates a `HistogramVec` with options the `opts` options
+    /// and the `label_names` label names.
```
Comment (queenypingcap) [inline src/histogram.rs]: /// `new` creates a `HistogramVec` with the `opts` options

File: src/vec.rs [hunk 15]
```diff
@@ -147,17 +149,19 @@ impl<T: MetricVecBuilder> MetricVecCore<T> {
...
 
-// MetricVec is a Collector to bundle metrics of the same name that
-// differ in their label values. MetricVec is usually not used directly but as a
-// building block for implementations of vectors of a given metric
-// type. GaugeVec, CounterVec, SummaryVec, and UntypedVec are examples already
-// provided in this package.
+/// `MetricVec` is a Collector to bundle metrics of the same name that
+/// differ in their label values. `MetricVec` is usually not used directly but as a
+/// building block for implementations of vectors of a given metric
+/// type. `GaugeVec`, `CounterVec`, `SummaryVec`, and `UntypedVec` are examples already
```
Comment (queenypingcap) [inline src/vec.rs]: The comma in this sentence should be changed to half-width characters.

**Comment by overvenus** [comment]
LGTM

**Pull request merged**